examples/3d/parallax_mapping.rs

```diff
@@ -0,0 +1,352 @@
+    };
+    let mut text = text.single_mut();
+    text.sections[2].value = format!("Algorithm: {algo_name}\n");
+
+    for (_, mat) in materials.iter_mut() {
+        mat.parallax_algorithm = *current;
+    }
+}
+
+fn update_parallax_layers(
+    input: Res<Input<KeyCode>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut target_layers: Local<TargetLayers>,
+    mut text: Query<&mut Text>,
+) {
+    if input.just_pressed(KeyCode::Key3) {
+        target_layers.0 -= 1.0;
+        target_layers.0 = target_layers.0.max(0.0);
+    } else if input.just_pressed(KeyCode::Key4) {
+        target_layers.0 += 1.0;
+    } else {
+        return;
+    }
+    let layer_count = target_layers.0.exp2();
+    let mut text = text.single_mut();
+    text.sections[1].value = format!("Layers: {layer_count:.0}\n");
+
+    for (_, mat) in materials.iter_mut() {
+        mat.max_parallax_layer_count = layer_count;
+    }
+}
+
+fn spin(time: Res<Time>, mut query: Query<(&mut Transform, &Spin)>) {
+    for (mut transform, spin) in query.iter_mut() {
+        transform.rotate_local_y(spin.speed * time.delta_seconds());
+        transform.rotate_local_x(spin.speed * time.delta_seconds());
+        transform.rotate_local_z(spin.speed * time.delta_seconds());
+    }
+}
+
+// Camera positions to cycle through when left-clickig.
+const CAMERA_POSITIONS: &[Transform] = &[
+    Transform {
+        translation: Vec3::new(1.5, 1.5, 1.5),
+        rotation: Quat::from_xyzw(-0.279, 0.364, 0.115, 0.880),
+        scale: Vec3::ONE,
+    },
+    Transform {
+        translation: Vec3::new(2.4, 0.0, 0.2),
+        rotation: Quat::from_xyzw(0.094, 0.676, 0.116, 0.721),
+        scale: Vec3::ONE,
+    },
+    Transform {
+        translation: Vec3::new(2.4, 2.6, -4.3),
+        rotation: Quat::from_xyzw(0.170, 0.908, 0.308, 0.225),
+        scale: Vec3::ONE,
+    },
+    Transform {
+        translation: Vec3::new(-1.0, 0.8, -1.2),
+        rotation: Quat::from_xyzw(-0.004, 0.909, 0.247, -0.335),
+        scale: Vec3::ONE,
+    },
+];
+
+fn move_camera(
+    mut camera: Query<&mut Transform, With<CameraController>>,
+    mut current_view: Local<usize>,
+    button: Res<Input<MouseButton>>,
+) {
+    let mut camera = camera.single_mut();
+    if button.just_pressed(MouseButton::Left) {
+        *current_view = (*current_view + 1) % CAMERA_POSITIONS.len();
+    }
+    let target = CAMERA_POSITIONS[*current_view];
+    camera.translation = camera.translation.lerp(target.translation, 0.2);
+    camera.rotation = camera.rotation.slerp(target.rotation, 0.2);
+}
+
+fn setup(
+    mut commands: Commands,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut normal: ResMut<Normal>,
+    asset_server: Res<AssetServer>,
+) {
+    // The normal map. Note that to generate it in the GIMP image editor, you should
+    // open the bump map, and do Filters → Generic → Normal Map
+    // You should enable the "flip X" checkbox.
```

Comment:
Why? This kind of thing could be the reason for the minus sign in the bitangent.

Reply:
`in.world_normal` used for the tangent transformation should be independent of the normal map. And the cube is just a `shape::Cube`, so I assume the vertex normals are generated correctly.

I didn't take the time to think about this more carefully, but I suspect it's a limitation of GIMPs normal map filter.